### ISSUE TITLE
Gemma 4 GPU decode garbage on Samsung Xclipse 530 (SOCL "Valid mapped region entry not found")

### ISSUE THREAD

**Issue by DenisovAV** [google-ai-edge/LiteRT-LM#2611]
### Summary

Gemma 4 E2B (`.litertlm`) on GPU produces garbage repeating output (multilingual / emoji token salad) on a **Samsung Galaxy A55 (Xclipse 530 GPU)**. CPU output is correct. The OpenCL delegate initializes and delegates all nodes (`LITERT_CL`), but during decode the Samsung OpenCL layer repeatedly logs `Valid mapped region entry not found`, in sync with the garbage tokens.

This looks **distinct from #1850** (`clEnqueueNDRangeKernel - Invalid command queue`, reported on Adreno): same end symptom (Gemma 4 GPU garbage) but a different failing operation — memory-region mapping, not command-queue.

### Device

- Samsung Galaxy A55 — **Xclipse 530** GPU (Exynos 1480, RDNA-based)
- OpenCL runs via **ANGLE → Vulkan** translation:
  - `ANGLE Renderer (Samsung Xclipse 530) on Vulkan 1.3.279`
  - `XGL SUMD Driver version: 24.0.545`
- Android; `litert-lm` 0.13.x (consumed via the `flutter_gemma` plugin)

### Model

`litert-community/gemma-4-E2B-it.litertlm`, GPU backend, `max_tokens=4096`, multimodal (`max_num_images=4`), speculative decoding off.

### Logs (decode phase)

```
litert  [gpu_environment.cc:217] Created OpenCL device from provided device id and platform id.
tflite  Replacing 2068 out of 2068 node(s) with delegate (LITERT_CL) node, yielding 1 partitions for subgraph 0.
ANGLE   Version (24.1.295 ...), Renderer ((Samsung Xclipse 530) on Vulkan 1.3.279)
native  delegate_kernel.cc:788 Initializing OpenCL-based API from serialized data.
...
SOCL  E  Valid mapped region entry not found        <-- repeats during decode
SOCL  E  Valid mapped region entry not found
flutter  Text accumulated: "设施📏ЭтаMayo👁 earthquakesRag 用户Lion石头📏 लोहाबर्ग国道बर्गRag..."   <-- garbage
```

`SOCL` here is the Samsung OpenCL stack. The `Valid mapped region entry not found` errors fire repeatedly during decode, time-correlated with the garbage tokens — suggesting the ANGLE/Vulkan-translated OpenCL fails to map the GPU memory regions the runtime expects, so decode reads/writes wrong memory and emits broken logits.

### Questions

1. Is the `SOCL: Valid mapped region entry not found` path a known issue on **Xclipse / ANGLE-translated OpenCL**?
2. Is there an engine flag (analogous to the #1850 `external_tensor_mode=false` / `hint_waiting_for_completion=true` workaround) that avoids the memory-region mapping the Samsung CL driver can't satisfy?
3. Or is a CPU fallback for decode the only reliable option on this GPU stack?

Happy to provide the full logcat or test patches. Original downstream report: DenisovAV/flutter_gemma#214.
